src/Microsoft.AspNet.Mvc.ModelBinding/ModelBindingContext.cs

```diff
@@ -71,12 +70,14 @@ public object Model
             set
             {
                 IsModelSet = true;
-
+                ModelStateKey = ModelName;
```

Comment:
I had something similar 
```public class ModelBindingResult
{
    public object Model {}
    public string Key {}
    public bool IsModelSet {}
}
```
but that seemed like not very intutive from a model binder perspective. But i am leaning towards doing this.